File: api/titik_meter.php
<?php
require_once __DIR__ . '/config.php';

$area_id = $_GET['area_id'] ?? null;

if (!empty($area_id)) {
    $stmt = $pdo->prepare("
        SELECT tm.*, a.nama AS area_nama 
        FROM titik_meter tm 
        LEFT JOIN area a ON tm.area_id = a.id 
        WHERE tm.status = 'aktif' AND tm.area_id = ? 
        ORDER BY tm.nama ASC
    ");
    $stmt->execute([$area_id]);
} else {
    $stmt = $pdo->query("
        SELECT tm.*, a.nama AS area_nama 
        FROM titik_meter tm 
        LEFT JOIN area a ON tm.area_id = a.id 
        WHERE tm.status = 'aktif' 
        ORDER BY tm.nama ASC
    ");
}

$rows = $stmt->fetchAll();
$data = array_map(function($tm) {
    return [
        'id' => (int)$tm['id'],
        'area_id' => (int)$tm['area_id'],
        'nama' => $tm['nama'],
        'lokasi_flow_meter' => $tm['lokasi_flow_meter'] ?? null,
        'meter_faktor' => (float)$tm['meter_faktor'],
        'tarif_harga' => (float)$tm['tarif_harga'],
        'status' => $tm['status'],
        'area_nama' => $tm['area_nama'] ?? '-'
    ];
}, $rows);

json_response(['data' => $data]);

File: flutter_app/lib/models/titik_meter.dart
class TitikMeter {
  final int id;
  final int areaId;
  final String nama;
  final String? lokasiFlowMeter;
  final double meterFaktor;
  final double tarifHarga;
  final String status;
  final String areaNama;

  TitikMeter({
    required this.id,
    required this.areaId,
    required this.nama,
    this.lokasiFlowMeter,
    required this.meterFaktor,
    required this.tarifHarga,
    required this.status,
    required this.areaNama,
  });

  factory TitikMeter.fromJson(Map<String, dynamic> json) {
    return TitikMeter(
      id: json['id'],
      areaId: json['area_id'],
      nama: json['nama'],
      lokasiFlowMeter: json['lokasi_flow_meter'],
      meterFaktor: (json['meter_faktor'] as num).toDouble(),
      tarifHarga: (json['tarif_harga'] as num).toDouble(),
      status: json['status'],
      areaNama: json['area_nama'] ?? '-',
    );
  }
}
